dbms/src/Core/Block.cpp

```diff
@@ -543,8 +543,12 @@ Block mergeBlocks(Blocks && blocks)
 
     for (size_t i = 1; i < blocks.size(); ++i)
     {
-        if (likely(blocks[i].rows()) > 0)
+        if (likely(blocks[i].rows() > 0))
         {
+            if (unlikely(!blocksHaveEqualStructure(first_block, blocks[i])))
```

Comment:
why the same block case will be encountered? Are there some expected cases that it needs merge identical blocks?

Reply:
> why the same block case will be encountered? Are there some expected cases that it needs merge identical blocks?

This is for protection, anyone can call mergeBlocks, if there is a different structure in the input blocks, the result will be problematic.  Its function is  similar here https://github.com/pingcap/tiflash/blob/fececd9375bd3b5de2318788bb08fa84e7c5deed/dbms/src/Interpreters/Aggregator.cpp#L2042

Reply:
> why the same block case will be encountered? Are there some expected cases that it needs merge identical blocks?

merging identical blocks you can refer https://github.com/pingcap/tiflash/issues/6533